cardano-integration/src/Test/Integration/Cardano/Local.hs

```diff
@@ -0,0 +1,444 @@
+import Data.Functor (void, (<&>))
+import Data.List (isInfixOf)
+import Data.Maybe (fromJust)
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import Data.Time (UTCTime, addUTCTime, getCurrentTime, nominalDiffTimeToSeconds, secondsToNominalDiffTime)
+import Data.Time.Clock.POSIX (utcTimeToPOSIXSeconds)
+import qualified Data.Yaml.Aeson as YAML
+import GHC.Generics (Generic)
+import System.FilePath ((</>))
+import System.IO (Handle, IOMode(..), hClose, openFile)
+import System.Process (CreateProcess(..), ProcessHandle, StdStream(..), cleanupProcess, createProcess, proc)
+import System.Random (randomRIO)
+import Test.Integration.Cardano.Process (execCli_)
+import Test.Integration.Workspace
+  ( Workspace(..)
+  , copyToWorkspace
+  , createWorkspace
+  , createWorkspaceDir
+  , moveToWorkspace
+  , resolveWorkspacePath
+  , rewriteJSONFile
+  , rewriteYAMLFile
+  , writeWorkspaceFileJSON
+  )
+import qualified Test.Integration.Workspace as W
+import Text.Printf (printf)
+import UnliftIO.Exception (catchAny)
+import UnliftIO.Resource (allocate, runResourceT, unprotect)
+
+data LocalTestnetOptions = LocalTestnetOptions
+  { slotDuration :: Int
+  , securityParam :: Int
+  , numSpoNodes :: Int
+  , numDelegators :: Int
+  , numWallets :: Int
+  , totalBalance :: Int
+  } deriving (Show, Eq, Ord)
+
+data LocalTestnet = LocalTestnet
+  { delegators :: [Delegator]
+  , network :: Network
+  , spoNodes :: [SpoNode]
+  , testnetMagic :: Int
+  , wallets :: [PaymentKeyPair]
+  , workspace :: Workspace
+  }
+
+data PaymentKeyPair = PaymentKeyPair
+  { paymentSKey :: FilePath
+  , paymentVKey :: FilePath
+  } deriving (Show, Eq, Ord, Generic)
+
+instance ToJSON PaymentKeyPair
+
+data StakingKeyPair = StakingKeyPair
+  { stakingSKey :: FilePath
+  , stakingVKey :: FilePath
+  } deriving (Show, Eq, Ord, Generic)
+
+instance ToJSON StakingKeyPair
+
+data Delegator = Delegator
+  { paymentKeyPair :: PaymentKeyPair
+  , stakingKeyPair :: StakingKeyPair
+  } deriving (Show, Eq, Ord, Generic)
+
+instance ToJSON Delegator
+
+data SpoNode = SpoNode
+  { byronDelegateKey :: FilePath
+  , byronDelegationCert :: FilePath
+  , coldSKey :: FilePath
+  , coldVKey :: FilePath
+  , db :: FilePath
+  , kesSKey :: FilePath
+  , kesVKey :: FilePath
+  , nodeName :: String
+  , opcert :: FilePath
+  , port :: Int
+  , processHandle :: ProcessHandle
+  , socket :: FilePath
+  , stakingRewardSKey :: FilePath
+  , stakingRewardVKey :: FilePath
+  , stderrLogs :: FilePath
+  , stdin :: Handle
+  , stdoutLogs :: FilePath
+  , topology :: FilePath
+  , vrfSKey :: FilePath
+  , vrfVKey :: FilePath
+  }
+
+data Network = Network
+  { configurationYaml :: FilePath
+  , byronGenesisJson :: FilePath
+  , shelleyGenesisJson :: FilePath
+  , alonzoGenesisJson :: FilePath
+  } deriving (Show, Eq, Ord, Generic)
+
+instance ToJSON Network
+
+defaultOptions :: LocalTestnetOptions
+defaultOptions = LocalTestnetOptions
+  { slotDuration = 1000
+  , securityParam = 10
```

Comment:
This might be too low for rollback testing.

Reply:
I is just a default - you can provide a different value.